ql/src/java/org/apache/hadoop/hive/ql/scheduled/ScheduledQueryExecutionService.java

```diff
@@ -73,41 +89,42 @@ public static ScheduledQueryExecutionService startScheduledQueryExecutorService(
 
   private ScheduledQueryExecutionService(ScheduledQueryExecutionContext ctx) {
     context = ctx;
-    ctx.executor.submit(worker = new ScheduledQueryExecutor());
+    ctx.executor.submit(poller = new ScheduledQueryPoller());
     ctx.executor.submit(new ProgressReporter());
   }
 
   static boolean isTerminalState(QueryState state) {
     return state == QueryState.FINISHED || state == QueryState.FAILED;
   }
 
-  class ScheduledQueryExecutor implements Runnable {
-
-    private ScheduledQueryProgressInfo info;
+  class ScheduledQueryPoller implements Runnable {
```

Comment:
Can we leave a comment with high level purpose of the Poller class? Also, probably mentioning that there should be only one-at-the-time per HS2.